extensions/amp-timeago/0.2/use-intersect.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {useLayoutEffect, useRef, useState} from '../../../src/preact';
+
+/**
+ * @param {object} ref
+ * @return {?Array<IntersectionObserverEntry>}
+ */
+export function useIntersect(ref) {
+  const {0: entries, 1: set} = useState([]);
+  const observerRef = useRef(null);
+  if (observerRef.current === null) {
+    observerRef.current = new IntersectionObserver(set);
+  }
+
+  useLayoutEffect(() => {
+    const {current: node} = ref;
```

Comment:
Nit: Add a comment explaining that this must be done in the callback, since `ref.current` will change between the time that `useIntersect` is called and this callback is called.

Reply:
It also needs to be done in a callback because we want this to rerun every time the ref is updated. The consumer of this hook could hypothetically change the observed node over time.